tools/draincluster/drain/drain.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package drain
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	k8errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils"
+	"go.goms.io/fleet/pkg/utils/condition"
+	evictionutils "go.goms.io/fleet/pkg/utils/eviction"
+	toolsutils "go.goms.io/fleet/tools/utils"
+)
+
+const (
+	drainEvictionNameFormat     = "drain-eviction-%s-%s"
+	resourceIdentifierKeyFormat = "%s/%s/%s/%s/%s"
+)
+
+type Helper struct {
+	HubClient                            client.Client
+	ClusterName                          string
+	ClusterResourcePlacementResourcesMap map[string][]placementv1beta1.ResourceIdentifier
+}
+
+func (h *Helper) Drain(ctx context.Context) (bool, error) {
+	if err := h.cordon(ctx); err != nil {
+		return false, fmt.Errorf("failed to cordon member cluster %s: %w", h.ClusterName, err)
+	}
+
+	if err := h.fetchClusterResourcePlacementToEvict(ctx); err != nil {
+		return false, err
+	}
+
+	if len(h.ClusterResourcePlacementResourcesMap) == 0 {
+		log.Printf("There are currently no resources propagated to %s from fleet using ClusterResourcePlacement resources", h.ClusterName)
+		return true, nil
```

Comment:
Hi Arvind! Now the cluster is cordoned for no reason; at least we need to remind user to uncordon this.

Reply:
A (much) better way is to put the check front I assume.

Reply:
Hi Arvind! As previously discussed, let's add a log output here mentioning that the cluster has been cordoned with a taint.

Reply:
Added reminder here https://github.com/Azure/fleet/pull/1060/commits/1885458165bc404dbc21df83c3e61dbca3913312#diff-470da8ac4426d765d81fc981231b957157d9af879602ea08dc24a786311a13aaR77